Test/Form/Widget/AbstractWidgetTestCase.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\CoreBundle\Test\Form\Widget;
+
+use Symfony\Bridge\Twig\Extension\FormExtension;
+use Symfony\Bridge\Twig\Extension\TranslationExtension;
+use Symfony\Bridge\Twig\Form\TwigRenderer;
+use Symfony\Bridge\Twig\Form\TwigRendererEngine;
+use Symfony\Bridge\Twig\Tests\Extension\Fixtures\StubFilesystemLoader;
+use Symfony\Bundle\FrameworkBundle\Tests\Templating\Helper\Fixtures\StubTranslator;
+use Symfony\Component\Form\FormExtensionInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\Form\Test\TypeTestCase;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+abstract class AbstractWidgetTestCase extends TypeTestCase
+{
+    /**
+     * @var FormExtensionInterface
+     */
+    private $extension;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $rendererEngine = new TwigRendererEngine(array(
+            'formatter.html.twig',
+            'form_div_layout.html.twig',
+        ));
+
+        if (class_exists('Symfony\Component\Security\Csrf\CsrfTokenManagerInterface')) {
+            $csrfManagerClass = 'Symfony\Component\Security\Csrf\CsrfTokenManagerInterface';
+        } else {
+            $csrfManagerClass = 'Symfony\Component\Form\Extension\Csrf\CsrfProvider\CsrfProviderInterface';
+        }
+
+        $renderer = new TwigRenderer($rendererEngine, $this->getMock($csrfManagerClass));
+
+        $this->extension = new FormExtension($renderer);
+
+        /*
+         * this is an ugly workaround for different build strategies and, possibly, different TwigBridge installation directories
+         */
+        $twigPaths = array_filter(array(
+            __DIR__.'/../../../vendor/symfony/twig-bridge/Symfony/Bridge/Twig/Resources/views/Form',
+            __DIR__.'/../../../vendor/symfony/twig-bridge/Resources/views/Form',
+            __DIR__.'/../../../vendor/symfony/symfony/src/Symfony/Bridge/Twig/Resources/views/Form',
+            __DIR__.'/../../../../../symfony/symfony/src/Symfony/Bridge/Twig/Resources/views/Form',
```

Comment:
Please document each line, explaining in which situation each can happen. If you don't know, simply remove the line, and it will be added later with a comment if really needed. BTW, I don't think running tests from a Symfony project should be supported.